src/raftstore/store/worker/split_check.rs

```diff
@@ -158,9 +177,11 @@ impl<C: Sender<Msg>> Runner<C> {
         }
     }
 
-    fn check_split(&mut self, region: &Region) {
-        let mut split_ctx = self.coprocessor
-            .new_split_check_status(region, &self.engine);
+    fn check_split(&mut self, task: Task) {
+        let region = &task.region;
+        let mut split_ctx =
+            self.coprocessor
+                .new_split_check_status(region, &self.engine, task.is_auto_split());
```

Comment:
Why not just pass the split type? It seems pointless to define an enum and convert it to bool again.

Reply:
We'd like to use the enum instead of bool until we need to support more split algorithms